internal/scm/azuredevops/azuredevops.go

```diff
@@ -0,0 +1,540 @@
+		lCopy := label
+		labels[i] = core.WebApiTagDefinition{
+			Active: &activeLabel,
+			Name:   &lCopy,
+		}
+	}
+	srn := PrependPrefixIfNeeded(newPR.Head) // Pass the value of newPR.Head
+	trn := PrependPrefixIfNeeded(newPR.Base)
+
+	requestArgs := git.CreatePullRequestArgs{
+		GitPullRequestToCreate: &git.GitPullRequest{
+			Title:         &prTitle,
+			Description:   &newPR.Body,
+			SourceRefName: &srn,
+			TargetRefName: &trn,
+			CompletionOptions: &git.GitPullRequestCompletionOptions{
+				DeleteSourceBranch: &removeSourceBranch,
+			},
+			Labels:    &labels,
+			Reviewers: &reviewers,
+			IsDraft:   &newPR.Draft,
+		},
+		RepositoryId: &prr.rid,
+		Project:      &prr.ownerName,
+	}
+
+	pr, err := a.gitClient.CreatePullRequest(ctx, requestArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	return pullRequest{
+		ownerName:  prr.ownerName,
+		repoName:   prr.name,
+		repoID:     prr.rid,
+		branchName: newPR.Head,
+		id:         *pr.PullRequestId,
+	}, nil
+}
+
+func (a *AzureDevOps) UpdatePullRequest(ctx context.Context, repo scm.Repository, pullReq scm.PullRequest, updatedPR scm.NewPullRequest) (scm.PullRequest, error) {
+	r := repo.(repository)
+	pr := pullReq.(pullRequest)
+
+	reviewerIDs, err := a.getUserIds(ctx, append(updatedPR.Reviewers, updatedPR.Assignees...))
+	if err != nil {
+		return nil, err
+	}
+
+	reviewers := make([]git.IdentityRefWithVote, len(reviewerIDs))
+	for i, reviewerID := range reviewerIDs {
+		rCopy := reviewerID
+		reviewers[i] = git.IdentityRefWithVote{
+			Id: &rCopy,
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	prTitle := updatedPR.Title
+	if updatedPR.Draft {
+		prTitle = "Draft: " + prTitle
+	}
+
+	activeLabel := true
+	labels := make([]core.WebApiTagDefinition, len(updatedPR.Labels))
+
+	for i, label := range updatedPR.Labels {
+		lCopy := label
+		labels[i] = core.WebApiTagDefinition{
+			Active: &activeLabel,
+			Name:   &lCopy,
+		}
+	}
+
+	srn := PrependPrefixIfNeeded(updatedPR.Head) // Pass the value of newPR.Head
+	trn := PrependPrefixIfNeeded(updatedPR.Base)
+
+	updateResult, err := a.gitClient.UpdatePullRequest(ctx, git.UpdatePullRequestArgs{
+		GitPullRequestToUpdate: &git.GitPullRequest{
+			Title:         &prTitle,
+			Description:   &updatedPR.Body,
+			SourceRefName: &srn,
+			TargetRefName: &trn,
+			Labels:        &labels,
+			Reviewers:     &reviewers,
+		},
+		RepositoryId:  &pr.repoID,
+		PullRequestId: &pr.id,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return pullRequest{
+		ownerName:  r.ownerName,
+		repoName:   r.name,
+		repoID:     r.rid,
+		branchName: updatedPR.Head,
+		id:         *updateResult.PullRequestId,
+	}, nil
+}
+
+func (a *AzureDevOps) getUserIds(ctx context.Context, usernames []string) ([]string, error) {
+	userIds := make([]string, len(usernames))
+
+	searchFilter := "General"
+
+	for i, username := range usernames {
+		uCopy := username
+		users, err := a.identityClient.ReadIdentities(ctx, identity.ReadIdentitiesArgs{
```

Comment:
Not exactly sure how this API works. But since this seems like a search, might the result

Reply:
I wasn't totally sure on that, so I did some testing with my own username at work and it only returns exact matches.  So, I think we're good here, but let me know if you feel differently.

Reply:
As long as it is verified, there should be no problem 😄 Just thought, based on the API design, that it might return more results.